datafusion/src/logical_plan/expr.rs

```diff
@@ -1871,6 +1920,21 @@ mod tests {
         assert!(maybe_expr.is_err());
     }
 
+    #[test]
```

Comment:
Here are a few examples: is `time` column has  timestamp data type, value 10 needs to also be a timestamp (in nanosecond in this case) to have the expression analysis works